src/TableFunctions/TableFunctionFormat.cpp

```diff
@@ -0,0 +1,107 @@
+
+#include <Processors/Executors/PullingPipelineExecutor.h>
+#include <Processors/Formats/IInputFormat.h>
+
+#include <QueryPipeline/Pipe.h>
+#include <QueryPipeline/QueryPipelineBuilder.h>
+
+#include <Storages/StorageValues.h>
+
+#include <TableFunctions/TableFunctionFormat.h>
+#include <TableFunctions/TableFunctionFactory.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int LOGICAL_ERROR;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+void TableFunctionFormat::parseArguments(const ASTPtr & ast_function, ContextPtr context)
+{
+    ASTs & args_func = ast_function->children;
+
+    if (args_func.size() != 1)
+        throw Exception("Table function '" + getName() + "' must have arguments", ErrorCodes::LOGICAL_ERROR);
+
+    ASTs & args = args_func.at(0)->children;
+
+    if (args.size() != 2)
+        throw Exception("Table function '" + getName() + "' requires 2 arguments: format and data", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    for (auto & arg : args)
+        arg = evaluateConstantExpressionOrIdentifierAsLiteral(arg, context);
+
+    format = args[0]->as<ASTLiteral &>().value.safeGet<String>();
+    data = args[1]->as<ASTLiteral &>().value.safeGet<String>();
+}
+
+ColumnsDescription TableFunctionFormat::getActualTableStructure(ContextPtr context) const
+{
+    auto read_buffer_creator = [&]()
+    {
+        return std::make_unique<ReadBufferFromString>(data);
+    };
+
+    return readSchemaFromFormat(format, std::nullopt, read_buffer_creator, context);
+}
+
+Block TableFunctionFormat::parseData(ColumnsDescription columns, ContextPtr context) const
+{
+    Block result;
+    for (const auto & name_and_type : columns.getAllPhysical())
+        result.insert({name_and_type.type->createColumn(), name_and_type.type, name_and_type.name});
+
+    auto read_buf = std::make_unique<ReadBufferFromString>(data);
+    auto input_format = context->getInputFormat(format, *read_buf, result, context->getSettingsRef().max_block_size);
+    QueryPipelineBuilder builder;
+    builder.init(Pipe(input_format));
+    auto pipeline = std::make_unique<QueryPipeline>(QueryPipelineBuilder::getPipeline(std::move(builder)));
+    auto reader = std::make_unique<PullingPipelineExecutor>(*pipeline);
+    reader->pull(result);
+
+    /// In case when data contains more then 1 block we read all blocks
+    /// and combine them all to one big block (this is considered a rare case).
+    Block new_block;
+    while (reader->pull(new_block))
+    {
+        for (size_t i = 0; i != result.columns(); ++i)
+        {
+            auto & result_column = result.getByPosition(i);
+            const auto & new_column = new_block.getByPosition(i);
+            auto mutable_column = IColumn::mutate(result_column.column);
+            mutable_column->insertManyFrom(*new_column.column, 0, new_column.column->size());
+            result_column.column = std::move(mutable_column);
+        }
+    }
```

Comment:
Is it the same as here?
I wonder why we don't have function for concatenating blocks, but actually it's only 5-10 LOC, so it's ok to duplicate it.
https://github.com/ClickHouse/ClickHouse/blob/02eb2fae1fd8d3c79fe79047a2c5973d1f8815a5/src/Interpreters/SortedBlocksWriter.cpp#L325-L340

Reply:
I will add a function for blocks concatination